app/src/main/java/com/duckduckgo/app/browser/autocomplete/BrowserAutoCompleteSuggestionsAdapter.kt

```diff
@@ -16,55 +16,49 @@
 
 package com.duckduckgo.app.browser.autocomplete
 
-import android.view.LayoutInflater
-import android.view.View
 import android.view.ViewGroup
 import androidx.annotation.UiThread
 import androidx.recyclerview.widget.RecyclerView
 import com.duckduckgo.app.autocomplete.api.AutoCompleteApi.AutoCompleteSuggestion
-import com.duckduckgo.app.browser.R
-import com.duckduckgo.app.browser.autocomplete.BrowserAutoCompleteSuggestionsAdapter.AutoCompleteViewHolder
-import com.duckduckgo.app.browser.autocomplete.BrowserAutoCompleteSuggestionsAdapter.AutoCompleteViewHolder.EmptySuggestionViewHolder
-import com.duckduckgo.app.browser.autocomplete.BrowserAutoCompleteSuggestionsAdapter.AutoCompleteViewHolder.SuggestionViewHolder
-import kotlinx.android.synthetic.main.item_autocomplete_suggestion.view.*
-import javax.inject.Inject
+import com.duckduckgo.app.browser.autocomplete.AutoCompleteViewHolder.EmptySuggestionViewHolder
 
-class BrowserAutoCompleteSuggestionsAdapter @Inject constructor(
+class BrowserAutoCompleteSuggestionsAdapter (
     private val immediateSearchClickListener: (AutoCompleteSuggestion) -> Unit,
     private val editableSearchClickListener: (AutoCompleteSuggestion) -> Unit
 ) : RecyclerView.Adapter<AutoCompleteViewHolder>() {
 
+    private val viewHolderFactoryMap: MutableMap<Int, SuggestionViewHolderFactory> = mutableMapOf()
     private val suggestions: MutableList<AutoCompleteSuggestion> = ArrayList()
 
-
-    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): AutoCompleteViewHolder {
-
-        val inflater = LayoutInflater.from(parent.context)
-
-        return if (suggestions.isEmpty()) {
-            EmptySuggestionViewHolder(inflater.inflate(R.layout.item_autocomplete_no_suggestions, parent, false))
-        } else {
-            SuggestionViewHolder(inflater.inflate(R.layout.item_autocomplete_suggestion, parent, false))
+    init {
```

Comment:
is this `init` block needed? we could move the initialisation of the array into its definition. e.g., 

    private val viewHolderFactoryMap: Map<Int, SuggestionViewHolderFactory> = mapOf(
        EMPTY_TYPE to EmptySuggestionViewHolderFactory(),
        SUGGESTION_TYPE to SearchSuggestionViewHolderFactory(),
        BOOKMARK_TYPE to BookmarkSuggestionViewHolderFactory()
    )

the advantage being that we don't need to make it a `mutable` map (which it isn't, really)

Reply:
Yes, we don't need to initiliase it there, could be done in the definition. Done!